ocp_utilities/infra.py

```diff
@@ -354,3 +357,65 @@ def create_icsp(icsp_name, repository_digest_mirrors):
     )
     icsp.deploy()
     return icsp
+
+
+def dict_base64_encode(_dict):
+    """Encoding dict in base64
+
+    Args:
+        _dict (dict): data dict to be encoded
+    Returns:
+        _dict encoded in base64 (str)
+    """
+    return base64.b64encode(json.dumps(_dict).encode("ascii")).decode("utf-8")
+
+
+def create_update_secret(secret_data_dict, name, namespace):
+    """Update existing secret or create a new secret; secret type - dockerconfigjson
+
+    Args:
+        secret_data_dict (dict): Secret data to be added/created,
+            example: {"auths":{
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>},
+                    ...,
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>}
+                    }
+                }
+        name (str): Secret name
+        namespace (str): Secret namespace
+
+    Returns:
+        Secret object
+    """
+    secret = Secret(name=name, namespace=namespace)
+    secret_key = ".dockerconfigjson"
+    auths_key = "auths"
+
+    if secret.exists:
+        old_secret_data_dict = json.loads(
+            base64.b64decode(secret.instance.data[secret_key])
+        )[auths_key]
+        old_secret_data_dict.update(secret_data_dict[auths_key])
+        secret_data_encoded = dict_base64_encode(
+            _dict={auths_key: old_secret_data_dict}
+        )
+
+        ResourceEditor(
+            patches={secret: {"data": {secret_key: secret_data_encoded}}}
+        ).update()
+
+        return secret
+
+    secret_data_encoded = dict_base64_encode(_dict=secret_data_dict)
+    secret = Secret(
```

Comment:
return Secret(....)

Reply:
applying `secret.res.update(...)` and `secret.deploy` instead of old method